File: deck.rb
require './card'

class Deck

  attr_reader :cards
  def initialize
    @cards = []
    Card::POSSIBLE_SUITS.each do |suit|
      Card::POSSIBLE_VALUES.each do |card_value|
        @cards << Card.new(card_value, suit)
      end
    end
  end

  def recreate
    @cards = []
    Card::POSSIBLE_SUITS.each do |suit|
      Card::POSSIBLE_VALUES.each do |card_value|
        @cards << Card.new(card_value, suit)
      end
    end
  end

  def shuffle
    @cards.shuffle!
  end

  def draw_first_card
    drawn_card = @cards.first
    @cards.delete(drawn_card)
    drawn_card
  end

  def add_card(card)
    @cards << card
  end
end

File: cardgame2/deck.py
import random
from card import Card

class Deck:
    def __init__(self):
        self.cards = []
        self.recreate()

    def recreate(self):
        self.cards = []
        for suit in Card.POSSIBLE_SUITS:
            for card_value in Card.POSSIBLE_VALUES:
                self.cards.append(Card(card_value, suit))

    def shuffle(self):
        random.shuffle(self.cards)

    def draw_first_card(self):
        if not self.cards:
            return None
        drawn_card = self.cards[0]
        self.cards.remove(drawn_card)
        return drawn_card

    def add_card(self, card):
        self.cards.append(card)